src/SDK/SmartDetectorsShared/Clients/AnalysisServicesFactory.cs

```diff
@@ -75,15 +74,20 @@ public AnalysisServicesFactory(IExtendedTracer tracer, IHttpClientWrapper httpCl
         /// <returns>The telemetry data client, that can be used to run queries on log analytics workspaces.</returns>
         public async Task<ITelemetryDataClient> CreateLogAnalyticsTelemetryDataClientAsync(IReadOnlyList<ResourceIdentifier> resources, CancellationToken cancellationToken)
         {
+            // Verify that there are no Application Insights resources
+            if (resources.Any(resource => resource.ResourceType == ResourceType.ApplicationInsights))
+            {
+                throw new TelemetryDataClientCreationException($"Telemetry client creation failed - resources shouldn't contain resources of the type {ResourceType.ApplicationInsights}");
```

Comment:
>TelemetryDataClientCreationException [](start = 26, length = 36)

naming - this is very general, it's preferable to name exceptions so the actual error is clear from the class name

my suggestion: InvalidResourcesTypesException

Reply:
or InvalidResourcesException so you can reuse the exception for the case where the resource list is empty

---
In reply to: [244680169](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/35#discussion_r244680169) [](ancestors = 244680169)